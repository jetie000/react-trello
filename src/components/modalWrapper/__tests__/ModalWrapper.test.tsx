import * as React from "react"
import { render, screen } from "@testing-library/react"
import "@testing-library/jest-dom"
import ModalWrapper, { ModalProps } from "@/components/modalWrapper/ModalWrapper"

const renderModal = (params: ModalProps) => render(<ModalWrapper {...params} />)

const id = "modalId"
const title = "Modal Title"
const size = "md"
const children = "Modal Content"

describe("ModalWrapper", () => {
  it("should render a modal with the given id, title, size and children", () => {
    const component = renderModal({ id, children, size, title })

    expect(screen.getByTestId(id)).toBeInTheDocument()
    expect(screen.getByText(title)).toBeInTheDocument()
    expect(screen.getByText(children)).toBeInTheDocument()
    expect(component).toMatchSnapshot()
  })
  it("should render a modal with a header containing the title and a close button", () => {
    renderModal({ id, children, size, title })

    const modalElement = screen.getByTestId(id)
    expect(modalElement).toBeInTheDocument()
    expect(modalElement).toHaveAttribute("id", id)
    expect(modalElement).toHaveAttribute("aria-labelledby", id + "Label")

    const modalTitleElement = screen.getByText(title)
    expect(modalTitleElement).toBeInTheDocument()

    const closeButtonElement = screen.getByLabelText("Close")
    expect(closeButtonElement).toBeInTheDocument()
  })
})
