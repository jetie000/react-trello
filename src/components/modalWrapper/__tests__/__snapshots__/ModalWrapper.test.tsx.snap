// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalWrapper should render a modal with the given id, title, size and children 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        aria-hidden="true"
        aria-labelledby="modalIdLabel"
        class="modal  fade"
        data-testid="modalId"
        id="modalId"
        role="dialog"
        tabindex="-1"
      >
        <div
          class="modal-dialog modal-dialog-centered modal-md"
          role="document"
        >
          <div
            class="modal-content"
          >
            <div
              class="modal-header"
            >
              <h5
                class="modal-title"
                id="modalIdLabel"
              >
                Modal Title
              </h5>
              <button
                aria-label="Close"
                class="btn-close"
                data-bs-dismiss="modal"
                type="button"
              />
            </div>
            <div
              class="modal-body"
            >
              Modal Content
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      aria-hidden="true"
      aria-labelledby="modalIdLabel"
      class="modal  fade"
      data-testid="modalId"
      id="modalId"
      role="dialog"
      tabindex="-1"
    >
      <div
        class="modal-dialog modal-dialog-centered modal-md"
        role="document"
      >
        <div
          class="modal-content"
        >
          <div
            class="modal-header"
          >
            <h5
              class="modal-title"
              id="modalIdLabel"
            >
              Modal Title
            </h5>
            <button
              aria-label="Close"
              class="btn-close"
              data-bs-dismiss="modal"
              type="button"
            />
          </div>
          <div
            class="modal-body"
          >
            Modal Content
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
